### PR TITLE
MDCT-1987 & 1953 - Combine databases

### PR DESCRIPTION
**Pull request by gmrabian**
## Description

NOTE: Due to updated dependencies please run `yarn` in the app-api folder especially, and others as needed

Completes [MDCT-1987](https://qmacbis.atlassian.net/browse/MDCT-1987) & [MDCT-1953](https://qmacbis.atlassian.net/browse/MDCT-1953)
Now that the app is built out more, we realized there was never a case where we used or needed `reportData` without also needing `report`. This PR combines the databases and greatly simplifies the access vectors.

This also saves time on the API call because it reduces 2 calls from the frontend down to 1 frontend call. The backend calls are also significantly reduced because they are not fetching across tables.

Please ask for walkthroughs or explanations as needed!

TABLES BEFORE:
reports
Screen Shot 2022-09-22 at 10 36 23 AM

report-data
Screen Shot 2022-09-22 at 10 36 31 AM

TABLES AFTER:
Reports and report-data combined:
Screen Shot 2022-09-22 at 10 38 05 AM

### How to test
Run locally. Nothing breaks. Note the changes below.

### Changed Dependencies
✨ root `.env` file change! See `.env_example` or `README.md` for more info ✨ 

NOTE: Due to updated dependencies please run `yarn` in the app-api folder especially, and others as needed

`MCPAR_REPORT_TABLE_NAME=local-mcpar-reports`

## Code author checklist
- [x] I have performed a self-review of my code
- [x] I have added [thorough](https://bit.ly/3zPrxuZ) tests
- [x] I have added analytics, if necessary
- [x] I have updated the documentation, if necessary

## Reviewer checklist (two different people)
- [ ] I have done the deep review and verified the items in the above checklist are g2g
- [ ] I have done the lgtm review

### CODE REVIEW

**Comment by gmrabian** [comment]
Status: 
1. Three failing ui-src test
2. Need to rewrite tests for read, create, update on `reports`
3. Hydration on `AddEditProgramModal` acting funky likely because of new data structure (fieldData nested within reports.)

Hydration in the report seems to work fine and all the other systems seem ok. Just need to fix these things and then refactor some more!

File: services/app-api/handlers/reports/update.ts [hunk 1]
```diff
@@ -42,12 +42,12 @@ export const updateReport = handler(async (event, context) => {
...
     const currentReport = JSON.parse(getCurrentReport.body);
     const { formTemplate } = currentReport;
-    const validatedFieldData = validateFieldData(
+    const validatedFieldData = await validateFieldData(
       formTemplate.validationJson,
       unvalidatedFieldData
     );
 
-    if (validatedMetadata && validatedFieldData) {
+    if (validatedMetadata) {
```
Comment (braxex) [inline services/app-api/handlers/reports/update.ts:50]: I might be missing something here, but why remove the check for validatedFieldData?

File: services/app-api/handlers/reports/update.ts [hunk 2]
```diff
@@ -57,7 +57,10 @@ export const updateReport = handler(async (event, context) => {
           state,
           id,
           lastAltered: Date.now(),
-          fieldData: validatedFieldData,
+          fieldData: {
+            ...currentReport.fieldData,
+            ...validatedFieldData,
+          },
```
Comment (braxex) [inline services/app-api/handlers/reports/update.ts:63]: Wouldn't this just write over the fieldData object with only what's being updated?

File: services/app-api/handlers/reports/update.ts (hunk 1 above)
Comment (gmrabian) [reply services/app-api/handlers/reports/update.ts:50]: Good call. Added an await to the above function and now it's covered - will add it back!

File: services/app-api/handlers/reports/update.ts (hunk 2 above)
Comment (gmrabian) [reply services/app-api/handlers/reports/update.ts:63]: spreading currentReport.fieldData takes care of that!

**Comment by ntsummers1** [comment]
Will get my review up in a second but I wonder if on big PRs like these if DB screenshots of the changes or a screen record of the difference would be helpful? Its something I pull up to compare for myself when reviewing but since I do it maybe others do to and that would help keep focus on what got changed?

**Review by braxex — APPROVED** [review]

**Comment by gmrabian** [comment]
@ntsummers1 good suggestion! I added screenshots of the db to the description.

**Review by BearHanded — APPROVED** [review]

**Review by ntsummers1 — APPROVED** [review]
I think this is good. Not seeing an issue on my end!

**Pull request merged**